src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java

```diff
@@ -58,7 +61,18 @@ public void onBlockPlaceExisting(BlockPlaceEvent e) {
         // This prevents Players from placing a block where another block already exists
         // While this can cause ghost blocks it also prevents them from replacing grass
         // or saplings etc...
-        if (BlockStorage.hasBlockInfo(e.getBlock())) {
+        Block block = e.getBlock();
+        if (BlockStorage.hasBlockInfo(block)) {
+            if (e.getBlockReplacedState().getType().isAir()) {
+                Iterator<ItemStack> item = BlockStorage.check(block).getDrops().iterator();
+                World world = block.getWorld();
+                Location loc = block.getLocation();
+                while (item.hasNext()) {
+                    world.dropItem(loc, item.next());
+                }
+                BlockStorage.clearBlockInfo(block);
+                return;
```

Comment:
Why would you `return;` here...
If this is to prevent the Event from being cancelled, use an `if/else` structure.